services/management/azure_database.go

```diff
@@ -0,0 +1,334 @@
+	client, err := s.getAzureClient(req)
+	if err != nil {
+		return nil, err
+	}
+
+	dataInstData, err := s.fetchAzureDatabaseInstancesData(ctx, req, client)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := managementpb.DiscoverAzureDatabaseResponse{}
+
+	for _, instance := range dataInstData {
+		inst := managementpb.DiscoverAzureDatabaseInstance{
+			Id:            instance.ID,
+			Location:      instance.Location,
+			Name:          instance.Name,
+			ResourceGroup: instance.ResourceGroup,
+			Environment:   instance.Tags["environment"],
+			Zones:         instance.Zones,
+		}
+		switch instance.Type {
+		case "microsoft.dbformysql/servers",
+			"microsoft.dbformysql/flexibleservers":
+			inst.Type = managementpb.DiscoverAzureDatabaseType_DISCOVER_AZURE_DATABASE_TYPE_MYSQL
+		case "microsoft.dbforpostgresql/servers",
+			"microsoft.dbforpostgresql/flexibleservers",
+			"microsoft.dbforpostgresql/serversv2":
+			inst.Type = managementpb.DiscoverAzureDatabaseType_DISCOVER_AZURE_DATABASE_TYPE_POSTGRESQL
+		case "microsoft.dbformariadb/servers":
+			inst.Type = managementpb.DiscoverAzureDatabaseType_DISCOVER_AZURE_DATABASE_TYPE_MARIADB
+		default:
+			inst.Type = managementpb.DiscoverAzureDatabaseType_DISCOVER_AZURE_DATABASE_TYPE_INVALID
+		}
+
+		if val, ok := instance.Properties["administratorLogin"].(string); ok {
+			inst.AdministratorLogin = fmt.Sprintf("%s@%s", val, instance.Name)
+		}
+		if val, ok := instance.Properties["fullyQualifiedDomainName"].(string); ok {
+			inst.FullyQualifiedDomainName = val
+		}
+
+		resp.AzureDatabaseInstance = append(resp.AzureDatabaseInstance, &inst)
+	}
+
+	return &resp, nil
+}
+
+// AddAzureDatabase add azure database to monitoring.
+//nolint:funlen
+func (s *AzureDatabaseService) AddAzureDatabase(ctx context.Context, req *managementpb.AddAzureDatabaseRequest) (*managementpb.AddAzureDatabaseResponse, error) {
+	l := logger.Get(ctx).WithField("component", "discover/azureDatabase")
+	if e := s.db.InTransaction(func(tx *reform.TX) error {
+		// tweak according to API docs
+		if req.NodeName == "" {
+			req.NodeName = req.InstanceId
+		}
+		if req.ServiceName == "" {
+			req.ServiceName = req.InstanceId
+		}
+
+		// tweak according to API docs
+		tablestatsGroupTableLimit := req.TablestatsGroupTableLimit
+		if tablestatsGroupTableLimit == 0 {
+			tablestatsGroupTableLimit = defaultTablestatsGroupTableLimit
+		}
+		if tablestatsGroupTableLimit < 0 {
+			tablestatsGroupTableLimit = -1
+		}
+
+		// add Remote Azure Database Node
+		node, err := models.CreateNode(tx.Querier, models.RemoteAzureDatabaseNodeType, &models.CreateNodeParams{
+			NodeName:     req.NodeName,
+			NodeModel:    req.NodeModel,
+			AZ:           req.Az,
+			Address:      req.InstanceId,
+			Region:       &req.Region,
+			CustomLabels: req.CustomLabels,
+		})
+		if err != nil {
+			return err
+		}
+		l.Infof("Created Azure Database Node with NodeID: %s", node.NodeID)
+
+		// add Azure Database Agent
+		if req.AzureDatabaseExporter {
```

Comment:
as far as I see we don't create any exporter if this flag is not passed and I'm not sure that it's what was expected.

Reply:
Agree. Removed.